packages/zero-runtime/README.md

```diff
@@ -0,0 +1,242 @@
+# zero-runtime
+
+A zero-runtime CSS-in-JS library that extracts the colocated css to it's own css files at build-time.
+
+## Getting started
+
+Zero-runtime supports Next.js and Vite with future support for more bundlers—you must install the corresponding plugin, as shown below.
+
+### Installation
+
+```bash
+npm install @mui/zero-runtime @mui/material
+# For Next.js
+npm install --save-dev @mui/zero-next-plugin
+# For Vite
+npm install --save-dev @mui/zero-vite-plugin
+```
+
+The package currently has a dependency on `@mui/material` to initialize the theme object, but this is only at build time. There won't be any Material UI code at runtime if you're not using it otherwise—in that case, you can move it to dev dependencies instead (as shown with the plugin packages).
+
+#### Next.js
+
+In your `next.config.js` file,
+
+1. Import the plugin
+
+```js
+const { withZeroPlugin } = require('@mui/zero-next-plugin');
+```
+
+2. Create a theme object
+
+```js
+const { experimental_extendTheme: extendTheme } = require('@mui/material/styles');
+const theme = extendTheme();
```

Comment:
What if we don't mention `@mui/material/styles` and use a plain theme object?

```suggestion
const theme = {
  colorSchemes: {
    light: {
      palette: {
        brand: '...',
      }
    },
    dark: {
      palette: {
        brand: '...',
      }
    }
  }
}
```

Reply:
We right now depend on the `extendTheme` implementation to then call `generateCssVars` in the `zero-runtime` util's `generateCss`. If we have to support plain objects, we can either implement the same in zero-runtime directly or we can move the theme implementation to `@mui/theming` package that will only have all theme specific logic.

Reply:
Got it. I think we need to revisit the theming for zero-runtime. In my opinion, zero-runtime should work without theme config.

If developer tries to use theme, we throw the error/warning and show them where to and the theme.

Next, the theme object can be a plain object that they can control. Meaning, if I provide:
```js
const theme = {
  typography: {
    h1: { … }, …
  }
}
```

I should be able to use it from the theme callback

Then the CSS variables, we should consider as an opt-in feature. Meaning, developers have to follow our structure and pass in `generateCssVars` to the theme.

I can take care of this task if you agree with the above model.

Reply:
Ideally, yes. It should work even with plain js objects. This was the case actually previously before we agreed on relying on making `extendTheme` the default. We can still provide css variable token object even if they are not using `extendTheme`. It would be possible if we extract out the css vars related utilities to either `@mui/system` or independent package.